### ISSUE TITLE
ifx-search-field: missing aria attributes

### ISSUE THREAD

**Issue by umang-sharma-ifx** [Infineon/infineon-design-system-stencil#2091]
### Summary

As a part of criteria **1.3.1**, and **4.1.2**, the combobox used in _ifx-search-field_ must have additional ARIA attributes to provide more accessibility information to screen-readers.

A developer customizable 'aria-controls' must be added to _ifx-search-field_ to help report what element the component influences. Also, 'aria-expanded' must be added to indicate if the combobox is expanded or collapsed. 

### Acceptance Criteria

- [ ] Meets technical requirements
- [ ] Meets design requirements
- [ ] Proper use of Semantic HTML
- [ ] Custom events are implemented
- [ ] Storybook integration
- [ ] Unit test integration
- [x] Meets accessibility requirements (Keyboarybord navigation, ARIA roles, etc.)
- [ ] Component has been thoroughly tested (Storybook, Angular, React, Vanilla JS and Vue)
- [ ] Component has been thoroughly reviewed

### Task Checklist

- [] Add an aria-controls attribute that developers can determine when invoking <_ifx-search-field>_.
- [] Add an aria-expanded attribute that developers can add to indicate if the component is expanded or collapsed.

### Notes

Required to meeting **1.3.1**, and **4.1.2** from [here](https://www.w3.org/WAI/WCAG21/Understanding/) to improve visual accessibility.
